Library/Formula/gtk+3.rb

```diff
@@ -22,6 +25,19 @@ class Gtkx3 < Formula
   depends_on "pango"
   depends_on "glib"
 
+  if build.with?("icons")
+    # Adwaita icon theme dependencies
+    depends_on "gettext" => :build
+    depends_on "icon-naming-utils" => :build
+    depends_on "intltool" => :build
+    depends_on "librsvg"
+  end
+
+  resource "adwaita-icon-theme" do
+    url "https://download.gnome.org/sources/adwaita-icon-theme/3.16/adwaita-icon-theme-3.16.2.1.tar.xz"
+    sha256 "b4556dfbf555d4fac12d4d5c12f7519de0d43ec42a1b649611439a50bf7acb96"
+  end if build.with?("icons")
```

Comment:
You don't need to protectively wrap resources. They won't be used unless invoked during `def install`.